src/server/rpc/procedures/ConnectN/ConnectN.js

```diff
@@ -0,0 +1,365 @@
+        open = this.board[row][column] === null,
+        isOnBoard = [row, column].every(this.isValidPosition.bind(this));
+
+
+    trace('"'+roleId+'" is trying to play at '+row+','+column+'. Board is \n'+
+        this.board.map(function(row) {
+            return row.map(t => t || '_').join(' ');
+    })
+    .join('\n'));
+    // Check that...
+
+    // ...the game is still going
+    if (this._winner) {
+        log('"'+roleId+'" is trying to play after the game is over');
+        return res.status(400).send(false);
+    }
+
+    // ...it's a valid position
+    if (!isOnBoard) {
+        log('"'+roleId+'" is trying to play in an invalid position ('+row+','+column+')');
+        return res.status(400).send(false);
+    }
+
+    // ...it is the given role's turn
+    if (this.lastMove === roleId) {
+        log('"'+roleId+'" is trying to play twice in a row!');
+        return res.status(400).send(false);
+    }
+
+    // ...it's not occupied
+    if (open) {
+        this.board[row][column] = roleId;
+        this._winner = ConnectNRPC.getWinner(this.board, this.numDotsToConnect);
+        trace('"'+roleId+'" successfully played at '+row+','+column);
+        // Send the play message to everyone!
+        req.netsbloxSocket._room.sockets()
+            .forEach(socket => socket.send({
+                type: 'message',
+                dstId: Constants.EVERYONE,
+                msgType: 'play',
+                content: {
+                    row: row,
+                    column: column,
+                    role: roleId
+                }
+            }));
+
+        this.lastMove = roleId;
+
+
+        trace('"'+roleId+'" is after playing at '+row+','+column+'. Board is \n'+
+            this.board.map(function(row) {
+                return row.map(t => t || '_').join(' ');
+            })
+                .join('\n'));
+
+
+        if(this.isGameOver())
+        {
+            req.netsbloxSocket._room.sockets()
+                .forEach(socket => socket.send({
+                type: 'message',
+                dstId: Constants.EVERYONE,
+                msgType: 'gameOver',
+                content: {
+                    winner: this._winner
+                }
+            }));
+        }
+
+        return res.status(200).send(true);
+    }
+    return res.status(400).send(false);
+};
+
+ConnectNRPC.prototype.isGameOver = function() {
+    var isOver = false;
+
+    // Game is over if someone has won
+    isOver = this._winner !== null;
+
+    // or all tiles are filled
+    isOver = isOver || this.isFullBoard();
+
+    log('isGameOver: ' + isOver + ' (' + this._winner + ')');
+    return isOver;
+};
+
+// Helper functions
+/**
+ * Get the winner from the given board layout.
+ *
+ * @param board
+ * @return {String} winner
+ */
+ConnectNRPC.getWinner = function(board, numDotsToConnect) {
+    var possibleWinners = [];
+    // Check for horizontal wins
+    //console.log("checking for horizontal");
```

Comment:
Either change these to use one of the loggers (`log`, `debug`, `trace` or `info`) or delete these commented out console logs